connector/spanmetricsconnectorv2/internal/aggregator/summary.go

```diff
@@ -0,0 +1,117 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package aggregator // import "github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2/internal/aggregator"
+
+import (
+	"time"
+
+	"github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2/config"
+	"github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2/internal/model"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// summary represents a summary metric which calculates count and sum over
+// span duration.
+type summary struct {
+	// TODO (lahsivjar): Attribute hash collisions are not considered
+	datapoints map[model.MetricKey]map[[16]byte]*summaryDP
+	timestamp  time.Time
+}
+
+func newSummary() *summary {
+	return &summary{
+		datapoints: make(map[model.MetricKey]map[[16]byte]*summaryDP),
+		timestamp:  time.Now(),
+	}
+}
```

Comment:
Yeah, this has been on my mind. I wanted to get all the data types first before making any effort there since I am not sure how the exponential histogram will come out to be.